### PR TITLE
feat(checkout): CHECKOUT-7281 Upgrade React 17

### PR DESCRIPTION
**Pull request by bc-peng**
## What?
Upgrade to React 17.

## Why?
This PR serves as an intermediate step toward upgrading `checkout-js` to React 18.

## Testing / Proof
image

@bigcommerce/checkout

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -164,6 +163,9 @@
     "webpack-inject-plugin": "^1.5.3",
     "yargs": "^14.0.0"
   },
+  "optionalDependencies": {
+    "@swc/core-linux-x64-gnu": "^1.2.160"
+  },
```
Comment (bc-peng) [inline package.json:168]: The CI complained about `Error: Cannot find module '@swc/core-linux-x64-gnu'`.
This is a solution from https://github.com/swc-project/swc/discussions/4152.

Comment (bc-peng) [inline package.json:168]: CircleCI complained about `Error: Cannot find module '@swc/core-linux-x64-gnu'`.
This is a solution from https://github.com/swc-project/swc/discussions/4152.

Comment (animesh1987) [reply package.json:168]: Do we know what this package actually does or why it's needed?

File: package.json [hunk 2]
```diff
@@ -56,9 +56,9 @@
     "@types/glob": "^7.2.0",
     "@types/googlemaps": "^3.30.16",
     "@types/lodash": "^4.14.136",
-    "@types/react": "^16.14.34",
-    "@types/react-datepicker": "^2.9.3",
-    "@types/react-dom": "^16.8.5",
+    "@types/react": "^17.0.2",
+    "@types/react-datepicker": "^4.10.0",
```
Comment (animesh1987) [inline package.json:60]: Did you test any changes with datepicker?

File: package.json (hunk 1 above)
Comment (bc-peng) [reply package.json:168]: It's binary for `@swc/core`, which is required by `nx`.
It is only missing in CI env, not in local.

**Review by animesh1987 — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (bc-peng) [reply package.json:60]: `react-datepicker` testing video:

https://user-images.githubusercontent.com/88361607/234486168-c529418f-8cae-4fcb-8f46-e20653966611.mov

Comment (animesh1987) [reply package.json:60]: Nicely done.

**Pull request merged**